### PR TITLE
Rename and more

### PR DESCRIPTION
**Pull request by sri2k1us**
- Allow users to rename file or folder.
- 
image

image

- Fix Create New Folder validation issue.
- Fix small `Login` link in banner for FF.
-Before:
image
-After:
image

### CODE REVIEW

File: src/components/data/toolbar/DataDotMenu.js [hunk 1]
```diff
@@ -369,6 +408,17 @@ function DataDotMenu(props) {
...
             />
+
+            {selectedResources && (
+                <RenameDialog
+                    path={selectedResources[0]?.path}
+                    open={renameDlgOpen}
+                    onClose={onRenameDlgClose}
+                    onRenamed={() => {
+                        onRenameDlgClose();
+                    }}
```
Comment (psarando) [inline src/components/data/toolbar/DataDotMenu.js:419]:
```suggestion
                    onRenamed={onRenameDlgClose}
```

File: src/components/data/RenameDialog.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                                        type="submit"
+                                        onClick={handleSubmit}
+                                    >
+                                        {t("rename")}
+                                    </Button>
+                                </>
+                            }
+                        >
+                            <Typography
+                                classes={{ root: classes.bottomPadding }}
```
Comment (psarando) [inline src/components/data/RenameDialog.js:118]: Is this different from setting `gutterBottom={true}`?
https://material-ui.com/api/typography/#props

**Review by psarando — APPROVED** [review]
LGTM :shipit: 👍

File: src/components/data/RenameDialog.js (hunk 2 above)
Comment (sri2k1us) [reply src/components/data/RenameDialog.js:118]: Hmm entirely sure but `gutterBottom` is for margins and this classes is setting `padding`. I copy-pasted from `CreateFolderDialog`.

**Comment by sri2k1us** [comment]
Thanks for the reivew. Additional reviews welcome.

File: src/components/data/RenameDialog.js (hunk 2 above)
Comment (psarando) [reply src/components/data/RenameDialog.js:118]: 👌 Well I just messed around with it in Storybook real quick, and they seem to have the same effect to me 🤷

Comment (sri2k1us) [reply src/components/data/RenameDialog.js:118]: May be we can get rid of `custom` style and replace it `gutterBottom` it in another PR from these 2 dialog 👍🏽

Comment (ash3rz) [reply src/components/data/RenameDialog.js:118]: Ah neat!  I either didn't know or forgot about `gutterBottom` 😄 I'll keep it in mind going forward

**Review by ash3rz — COMMENTED** [review]
LGTM :shipit: 
Not critical, but looks like the Rename option didn't make it into the row dot menu.

**Pull request merged**